src/qutip_tensorflow/core/data/norm.py

```diff
@@ -0,0 +1,69 @@
+import qutip
+from .tftensor import TfTensor
+import warnings
+
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=DeprecationWarning)
+    import tensorflow as tf
+
+
+def frobenius_tftensor(matrix):
+    return tf.norm(matrix._tf)
+
+
+def l2_tftensor(matrix):
+    if matrix.shape[0] != 1 and matrix.shape[1] != 1:
+        raise ValueError("L2 norm is only defined on vectors")
+    return frobenius_tftensor(matrix)
+
+
+def trace_tftensor(matrix):
+    # For column and row vectors we simply take frobenius norm as they are
+    # equivalent
+    if matrix.shape[0] == 1 or matrix.shape[1] == 1:
+        return frobenius_tftensor(matrix)
+
+    out = matrix._tf
+    out = tf.matmul(matrix._tf, matrix._tf, adjoint_b=True)
+    out = tf.linalg.sqrtm(out)
+    out = tf.linalg.trace(out)
+    return tf.math.real(out)
+
+
+def one_tftensor(matrix):
+    return tf.norm(matrix._tf, ord=1, axis=[-2, -1])
```

Comment:
Seems strange to use negative axes here, given that you presumably know there'll be only two anyway.  Is there a difference between doing `axis=[-2, -1]` and `axis=[0, 1]`?

Reply:
I think there is not difference for matrices but there will be a difference for matrix batches. The negative axes are suggested in tensorflow's documentation and I thin this is so that the norm for batches of MxN matrices represented as tensors of shape [A1, A2,..., M,N] can be computed with the same code. I would prefer not changing it just so that it is easier in the future to support batched operations if we want to. That being said, supporting batched operations would still require some changes in the rest of the code.